src/main/java/seedu/address/logic/commands/EditCommand.java

```diff
@@ -184,9 +197,18 @@ private CommandResult executeEditPerson(Model model, Person personToEdit) throws
             throw new CommandException(String.format(MESSAGE_DUPLICATE_PHONE, editedPerson.getPhone()));
         }
 
+        if (editPersonDescriptor.hasStudentTag()) {
+            editedPerson.removeTag("student");
+        }
+
         model.setPerson(personToEdit, editedPerson);
         model.updateFilteredPersonList(PREDICATE_SHOW_ALL_PERSONS);
 
+        if (editPersonDescriptor.hasStudentTag()) {
+            return new CommandResult(String.format(MESSAGE_EDIT_PERSON_SUCCESS + "\n" + MESSAGE_STUDENT_TAG_NOT_ADDED,
```

Comment:
in this case, edit command still proceeds even if user tries to add "student" tag to non-student so should we do the same for add command? i.e. if user tries to add "student" tag, we just remove it and let the command proceed? 
not sure if that can be done and not give bugs

Reply:
will make the change